crates/allocator/src/bump.rs

```diff
@@ -194,6 +195,18 @@ impl InnerAlloc {
             Some(alloc_start)
         }
     }
+
+    /// Aligns the start pointer of the next allocation.
+    ///
+    /// We inductively calculate the start index
+    /// of a layout in the linear memory.
+    /// - Initially `self.next` is `0`` and aligned
+    /// - `layout.align() - 1` accounts for `0` as the first index.
+    /// - the binary with the inverse of the align ensures
+    /// that the next allocated pointer address is of the power of 2.
```

Comment:
The inverse of the expression (align - 1) creates a bitmask that is used to zero out bits, ensuring alignment according to type requirements.

Reply:
- The expression `self.next + layout.align() - 1` ensures that the pointer is at least as large as the next alignment boundary.
- The bitwise AND with `!(layout.align() - 1)` clears the lower bits of the result to ensure that the address is a multiple of the alignment. The `!` operator creates a mask that zeroes out the lower bits that should not contribute to the final aligned address.